src/main/java/com/esri/samples/search/offline_geocode/OfflineGeocodeSample.java

```diff
@@ -0,0 +1,385 @@
+import com.esri.arcgisruntime.symbology.PictureMarkerSymbol;
+import com.esri.arcgisruntime.tasks.geocode.GeocodeParameters;
+import com.esri.arcgisruntime.tasks.geocode.GeocodeResult;
+import com.esri.arcgisruntime.tasks.geocode.LocatorTask;
+import com.esri.arcgisruntime.tasks.geocode.ReverseGeocodeParameters;
+
+public class OfflineGeocodeSample extends Application {
+
+  private MapView mapView;
+  private ArcGISMap map;
+  private ArcGISTiledLayer tiledLayer;
+  private LocatorTask locatorTask;
+  private GeocodeParameters geocodeParameters;
+  private ReverseGeocodeParameters reverseGeocodeParameters;
+  private GraphicsOverlay graphicsOverlay;
+  private PictureMarkerSymbol pinSymbol;
+  private ComboBox<String> searchBox;
+  private ListenableFuture<IdentifyGraphicsOverlayResult> identifyResults;
+  private boolean realtimeMode = false;
+
+  @Override
+  public void start(Stage stage) throws Exception {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Offline Geocode Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create search box
+      searchBox = new ComboBox<>();
+      searchBox.setPromptText("Search");
+      searchBox.setEditable(true);
+      searchBox.setMaxWidth(260.0);
+
+      // add example address suggestions
+      String[] recent = {
+          "1455 Market St, San Francisco, CA 94103", "2011 Mission St, San Francisco  CA  94110",
+          "820 Bryant St, San Francisco  CA  94103", "1 Zoo Rd, San Francisco, 944132",
+          "1201 Mason Street, San Francisco, CA 94108", "151 Third Street, San Francisco, CA 94103",
+          "1050 Lombard Street, San Francisco, CA 94109"
+      };
+      searchBox.getItems().addAll(recent);
+
+      // event to get geocode when query is submitted
+      searchBox.setOnAction((e) -> {
+
+        // get the user's query
+        String query = "";
+        if (searchBox.getSelectionModel().getSelectedIndex() == -1) {
+          // user supplied their own query
+          query = searchBox.getEditor().getText();
+        } else {
+          // user chose a suggested query
+          query = searchBox.getSelectionModel().getSelectedItem();
+        }
+
+        if (!query.equals("")) {
+          // hide callout if showing
+          mapView.getCallout().dismiss();
+
+          // run the locatorTask geocode task
+          ListenableFuture<List<GeocodeResult>> results = locatorTask.geocodeAsync(query, geocodeParameters);
+
+          // add a listener to display the result when loaded
+          results.addDoneListener(new ResultsLoadedListener(results));
+        }
+      });
+
+      // create a basemap from a local tile package
+      TileCache tileCache = new TileCache(System.getProperty("user.dir") +
```

Comment:
Could use code below since that is how we usually do it.
```java
final String tileCachePath = new File("samples-data/sanfrancisco/SanFrancisco.tpk").getAbsolutePath();
  TileCache tileCache = new TileCache(tileCachePath);
```